strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/domain/ArtifactEntry.java

```diff
@@ -18,7 +21,10 @@
     private String repositoryId;
 
     // if you have to rename this field please update ArtifactEntryServiceImpl.findByCoordinates() implementation
-    private ArtifactCoordinates artifactCoordinates;
+    @OneToOne(orphanRemoval = true)
+    private AbstractArtifactCoordinates artifactCoordinates;
+    
+    private String artifactPath;
```

Comment:
@carlspring you may want to note this denoramalization. This is needed to create unique index for ArtifactEntry.  The index in turn is needed in order to quickly determine the presence of the required entity.

Reply:
I was not able to find the best way :(